### ISSUE TITLE
Seeing negative values for wlentime and rlentime

### ISSUE THREAD

**Issue by sevagh** [zfsonlinux/zfs#8131]
### System information

Type | Version/Name
 --- | --- 
Distribution Name	| Debian
Distribution Version	| 8 (jessie)
Linux Kernel	| 3.16
Architecture	| amd64
ZFS Version	| 0.6.5.7-8-jessie

### Describe the problem you're observing

In kstat io stats for my zpool, there are negative values for wlentime and rlentime:

```
$ cat /proc/spl/kstat/zfs/ztank/io
20 3 0x00 1 80 16996408425 20115450070576800
nread    nwritten reads    writes   wtime    wlentime wupdate  rtime    rlentime rupdate  wcnt     rcnt
496595193856 326928208207872 28635533 1225798681 4589967829674802 -4516494086296980576 20115449382517651 4693467351523302 -4516494086296980576 20115449392170545 0        0
```

Not sure how to proceed from here - `/sbin/zpool status` shows everything is healthy, `dmesg` is blank and as far as I can tell the host and drives are healthy.

Is there any way to tell what's going on here?

**Comment by sevagh** [comment]
Found the same on one other machine:
```
$ cat /proc/spl/kstat/zfs/ztank/io
20 3 0x00 1 80 13807758822 39324853386226331
nread    nwritten reads    writes   wtime    wlentime wupdate  rtime    rlentime rupdate  wcnt     rcnt
6984923939840 84159544228352 364849642 3673837066 6942686870096519 -8944313648229711095 39324853372114993 7440786322703773 -8944313648229711095 39324853372116067 1583     4
```
It seems uncommon, it's only these 2 (out of thousands).

Good sign for its reproducability potential?

**Comment by behlendorf** [comment]
These values are only used for statistics tracking, so there's nothing wrong with the pool.  We'll take a look at this when we can.

Related to this `rlentime` is incorrectly reported in the old code as `wlentime`, you can cherry pick the fix if needed e8474f9ad3b3d23c3277535c4f53f8fd1e6cbd74.

**Comment by behlendorf** [comment]
@richardelling can you point us to the proposed fix.
